src/main/java/com/internet/shop/dao/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,190 @@
+            preparedStatement.setLong(1, shoppingCart.getUserId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                shoppingCart.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataBaseProcessingException("Shoping cart "
+                    + shoppingCart + " was not created", e);
+        }
+        return shoppingCart;
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        ShoppingCart shoppingCart = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM shoping_cart "
+                    + "WHERE id_shoping_cart = ? AND deleted = FALSE";
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                shoppingCart = new ShoppingCart();
+                shoppingCart.setId(id);
+                shoppingCart.setUserId(resultSet.getLong("id_user"));
+                shoppingCart.setProducts(getCartProducts(shoppingCart.getId()));
+                return Optional.of(shoppingCart);
+            }
+        } catch (SQLException e) {
+            throw new DataBaseProcessingException("Shoping cart "
+                    + shoppingCart + " was not created", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        List<ShoppingCart> shoppingCarts = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM shoping_cart WHERE deleted = FALSE";
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                ShoppingCart shoppingCart = new ShoppingCart();
+                shoppingCart.setId(resultSet.getLong("id_shoping_cart"));
+                shoppingCart.setUserId(resultSet.getLong("id_user"));
+                shoppingCart.setProducts(getCartProducts(shoppingCart.getId()));
+                shoppingCarts.add(shoppingCart);
+            }
+        } catch (SQLException e) {
+            throw new DataBaseProcessingException("Shopping cart list was not created", e);
+        }
+        return shoppingCarts;
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart shoppingCart) {
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "UPDATE shoping_cart "
+                    + "SET id_user = ? WHERE id_shoping_cart=? "
+                    + "AND deleted = FALSE";
```

Comment:
```suggestion
            String query = "UPDATE shoping_cart SET id_user = ?"
            + " WHERE id_shoping_cart=? AND deleted = FALSE";
```

Reply:
fixed query in update methop ShoppingCartDaoJdbcImpl